addons/sale_timesheet/controllers/main.py

```diff
@@ -12,74 +12,87 @@ class SaleTimesheetController(http.Controller):
+        dashboard_data = request.env['account.analytic.line'].read_group(dashboard_domain, ['unit_amount', 'timesheet_revenue', 'timesheet_invoice_type'], ['timesheet_invoice_type'])
         dashboard_total_hours = sum([data['unit_amount'] for data in dashboard_data])
         for data in dashboard_data:
             billable_type = data['timesheet_invoice_type']
-            # hours
             dashboard_values['hours'][billable_type] = float_round(data.get('unit_amount'), precision_rounding=hour_rounding)
             dashboard_values['hours']['total'] += float_round(data.get('unit_amount'), precision_rounding=hour_rounding)
             # rates
             rate = round(data.get('unit_amount') / dashboard_total_hours * 100, 2) if dashboard_total_hours else 0.0
             dashboard_values['rates'][billable_type] = rate
             dashboard_values['rates']['total'] += rate
 
-        # profitability
-        project_ids = values['timesheet_lines'].mapped('project_id').ids
-        money_amount = dict.fromkeys(['invoiced', 'to_invoice', 'cost', 'total'], 0.0)
-
-        profitability_raw_data = request.env['project.profitability.report.analysis'].read_group([('project_id', 'in', project_ids)], ['project_id', 'amount_untaxed_to_invoice', 'amount_untaxed_invoiced', 'timesheet_cost'], ['project_id'])
+        # profitability, using profitability SQL report
+        profit = dict.fromkeys(['invoiced', 'to_invoice', 'cost', 'total'], 0.0)
+        profitability_raw_data = request.env['project.profitability.report.analysis'].read_group([('project_id', 'in', projects.ids)], ['project_id', 'amount_untaxed_to_invoice', 'amount_untaxed_invoiced', 'timesheet_cost'], ['project_id'])
         for data in profitability_raw_data:
-            money_amount['invoiced'] += data.get('amount_untaxed_invoiced', 0.0)
-            money_amount['to_invoice'] += data.get('amount_untaxed_to_invoice', 0.0)
-            money_amount['cost'] += data.get('timesheet_cost', 0.0)
-        money_amount['total'] = sum([money_amount[item] for item in money_amount.keys()])
-        dashboard_values['money_amount'] = money_amount
+            profit['invoiced'] += data.get('amount_untaxed_invoiced', 0.0)
+            profit['to_invoice'] += data.get('amount_untaxed_to_invoice', 0.0)
+            profit['cost'] += data.get('timesheet_cost', 0.0)
+        profit['total'] = sum([profit[item] for item in profit.keys()])
+        dashboard_values['profit'] = profit
 
         values['dashboard'] = dashboard_values
 
-        # -- Time Repartition (per employee)
-        repartition_domain = domain + [('employee_id', '!=', False), ('timesheet_invoice_type', '!=', False)]  # force billable type
+        #
+        # Time Repartition (per employee per billable types)
+        #
+        employees = projects.mapped('tasks.user_id.employee_ids') | request.env['account.analytic.line'].search([('project_id', 'in', projects.ids)]).mapped('employee_id')
```

Comment:
[Commit3] Small perf proposal here: is the compute of employees based on aal really necessary ? Here below in repartition_employee you do a setdefault. I am therefore not sure it is necessary to compute the whole employees. Just having those coming from tasks in order to have them listed even if they made no timesheet seems sufficient.

Reply:
Unfortunately for performance, the Overview must contain all employee assigned to task, and all employee having timesheeted in the project ...
In the spec, it says "Time by people : Display all the users that have been assigned to tasks (time = 0 if no timesheet) TO CHECK : DOES IT ALSO DISPLAY THE EMPLOYEES WHO DID SOME TIMESHEETS ON THE TASKS ????  yes it does, nice job "

Reply:
NICE JOB !